pkg/apiserver/api_installer.go

```diff
@@ -390,18 +390,26 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 
 		resourcePath := namespacedPath
 		resourceParams := namespaceParams
+		itemPathPrefix := gpath.Join(a.prefix, scope.ParamName()) + "/"
 		itemPath := namespacedPath + "/{name}"
+		itemPathMiddle := "/" + resource + "/"
 		nameParams := append(namespaceParams, nameParam)
 		proxyParams := append(nameParams, pathParam)
+		itemPathSuffix := ""
 		if hasSubresource {
-			itemPath = itemPath + "/" + subresource
+			itemPathSuffix = "/" + subresource
+			itemPath = itemPath + itemPathSuffix
 			resourcePath = itemPath
 			resourceParams = nameParams
 		}
 		apiResource.Name = path
 		apiResource.Namespaced = true
 		apiResource.Kind = resourceKind
-		namer := scopeNaming{scope, a.group.Linker, gpath.Join(a.prefix, itemPath), false}
+
+		itemPathFn := func(name, namespace string) string {
+			return itemPathPrefix + namespace + itemPathMiddle + name + itemPathSuffix
```

Comment:
It's in a different branch.

On Mon, May 23, 2016 at 5:25 AM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In pkg/apiserver/api_installer.go
> <https://github.com/kubernetes/kubernetes/pull/26007#discussion_r64192201>
> :
>
> >  			resourcePath = itemPath
> >  			resourceParams = nameParams
> >  		}
> >  		apiResource.Name = path
> >  		apiResource.Namespaced = true
> >  		apiResource.Kind = resourceKind
> > -		namer := scopeNaming{scope, a.group.Linker, gpath.Join(a.prefix, itemPath), false}
> > +
> > +		itemPathFn := func(name, namespace string) string {
> > +			return itemPathPrefix + namespace + itemPathMiddle + name + itemPathSuffix
>
> What if the resource is non-namespaced (e.g. Node)?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26007/files/25a2a7d25a45edffcbdaefc20c93822fc71637e0..f670cc465211fb5328ee2beae330a6625750471c#r64192201>
>